### PR TITLE
Microsoft Dynamics 365 v2.4

### PR DESCRIPTION
**Pull request by dhirsch1**

### CODE REVIEW

File: microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc [hunk 1]
```diff
@@ -0,0 +1,934 @@
...
+| Connection a| * <<dynamics-config_connection, Connection>>
+* <<dynamics-config_oauth-user-pass, OAuth Username Password>>
+ | The connection types that can be provided to this configuration. | | x
+| Expiration Policy a| <<ExpirationPolicy>> | Configures the minimum amount of time that a dynamic configuration instance can remain idle before Mule considers it eligible for expiration. This does not mean that the instance expires at the exact moment that it becomes eligible. Mule purges the instances as appropriate. | |
+<<<<<<< Updated upstream
+| Encoding a| | The default body encoding type. If the encoding type is not specified at the operation level, this encoding type is used. | |
+=======
+| Encoding a| String | The default body encoding type. If the encoding type is not specified at the operation level, this encoding type is used. | |
+>>>>>>> Stashed changes
+|===
```
Comment (irinel-dan) [inline microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc:25]: >>>>>> Stashed changes - this looks like an unresolved merge conflict.

File: microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-xml-maven.adoc [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+The Maven pom.xml file generated by Anypoint Studio contains dependencies for
+a Mule app. When you code a Mule app manually, include this XML snippet in
+your pom.xml file to enable access to the connector. The Mule runtime engine
+translates the RELEASE dependency keyword to the current version.
+
+[source,xml,linenums]
+----
+<dependency>
+  <groupId>com.mulesoft.connectors</groupId>
+  <artifactId>mule-sap-connector</artifactId>
```
Comment (irinel-dan) [inline microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-xml-maven.adoc]: this should be
`<artifactId>mule-microsoft-dynamics365-connector</artifactId>`

**Review by irinel-dan — CHANGES_REQUESTED** [review]

**Review by irinel-dan — DISMISSED** [review]

**Review by catalinbarakat — DISMISSED** [review]

File: microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-studio.adoc [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+:page-aliases: connectors::microsoft/microsoft-dynamics-365-connector-design-center.adoc
+
+To configure a connector in Anypoint Studio:
+
+. Install the connector
+. Configure the connector
+. Configure an input source for the connector
+
+== Add the Connector to Your Project
+
```
Comment (valkyrie69) [inline microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-studio.adoc:12]: This is the old template. We should make a list of connectors that need to be updated to match the current template.

File: microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-xml-maven.adoc [hunk 4]
```diff
@@ -0,0 +1,78 @@
+= Microsoft Dynamics 365 Connector 2.4 XML and Maven Support - Mule 4
+:page-aliases: connectors::microsoft/microsoft-dynamics-365-connector-xml-maven.adoc
+
+If you manually code a Mule app in XML either from the Anypoint Studio XML editor or from a text editor, you can get access to the Microsoft Dynamics 365 Connector in your app by adding reference statements to your XML Mule flow and in the Apache Maven pom.xml file.
```
Comment (valkyrie69) [inline microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-xml-maven.adoc:4]: Outdated template

File: microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc [hunk 5]
```diff
@@ -0,0 +1,930 @@
...
+| Password a| String |  Password used to authenticate against the proxy. |  |
+| Reconnection a| <<Reconnection>> |  When the application is deployed, a connectivity test is performed on all connectors. If set to `true`, deployment fails if the test doesn't pass after exhausting the associated reconnection strategy. |  |
+| Pooling Profile a| <<PoolingProfile>> |  Characteristics of the connection pool |  |
+| Consumer Key a| String |  The OAuth key registered with the service provider |  | x
+| Consumer Secret a| String |  The OAuth consumer secret registered with the service provider |  | x
+| Authorization Url a| String |  The service provider's authorization endpoint URL |  `+https://login.microsoftonline.com/oauth2/authorize+` |
+| Access Token Url a| String |  The service provider's accessToken endpoint URL |  `+https://login.microsoftonline.com/oauth2/token+` |
+| Scopes a| String |  The OAuth scopes to be requested during the dance. If not provided, this value defaults to the scopes in the annotation. |  |
```
Comment (valkyrie69) [inline microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc]:
```suggestion
| Scopes a| String |  The OAuth scopes to request during the dance. If not provided, this value defaults to the scopes in the annotation. |  |
```

File: microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc [hunk 6]
```diff
@@ -0,0 +1,930 @@
...
+| Resource Owner Id a| String |  The ID each component should use if it doesn't have another reference. |  |
```
Comment (valkyrie69) [inline microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc:55]: What is "it"? 

The ID each component should use if the component doesn't have a reference to a resource owner ID? 

I'm not sure.

File: microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc [hunk 7]
```diff
@@ -0,0 +1,930 @@
...
+| Before a| String |  The name of a flow to execute right before starting the OAuth dance. |  |
```
Comment (valkyrie69) [inline microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc]:
```suggestion
| Before a| String |  The name of a flow to execute immediately before starting the OAuth dance. |  |
```

File: microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc [hunk 8]
```diff
@@ -0,0 +1,930 @@
...
+| After a| String |  The name of a flow to execute right after an accessToken is received. |  |
```
Comment (valkyrie69) [inline microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc]:
```suggestion
| After a| String |  The name of a flow to execute immediately after an accessToken is received. |  |
```

File: microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc [hunk 9]
```diff
@@ -0,0 +1,930 @@
...
+
+[%header%autowidth.spread]
+|===
+| Name | Type | Description | Default Value | Required
+| Version a| String | The version number of the API to which to connect  |  9.0 |
+| Read Timeout a| Number |  Specifies the amount of time, in milliseconds, that the consumer waits for a response before it times out. Default value is 0, which means infinite. |  0 |
+| Connection Timeout a| Number |  Specifies the amount of time, in milliseconds, for which the consumer attempts to establish a connection before the connection times out. Default value is 0, which means infinite. |  0 |
+| Host a| String |  Hostname of the proxy. If this property is not set, then no proxy will be used; otherwise, a proxy will be used, but a proxy host must be specified. |  |
+| Port a| Number |  Port of the proxy. If host is set, then this property must be set and cannot be a negative number. |  |
+| Username a| String |  Username used to authenticate against the proxy. If this property is not set then no authentication is going to use against the proxy, otherwise this value must be specified. |  |
```
Comment (valkyrie69) [inline microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc]:
```suggestion
| Username a| String |  Username used to authenticate against the proxy. If this property is not set, then no authentication is used against the proxy.  |  |
```

File: microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc [hunk 10]
```diff
@@ -0,0 +1,930 @@
...
+
+==== Parameters
+
+[%header%autowidth.spread]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | The name of the configuration to use. | | x
+| Logical Name a| String |  The logical name of the entity. It is always the schema name in lower case. |  | x
+| Ids a| Array of String |  The list of IDs to delete. |  `#[payload]` |
+| Use Single Transaction a| Boolean |  Use a single transaction for deletion. If the deletion fails, the transaction is rolled-back. |  true |
```
Comment (valkyrie69) [inline microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc]:
```suggestion
| Use Single Transaction a| Boolean |  Use a single transaction for deletion. If the deletion fails, the transaction is rolled back. |  true |
```

File: microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc [hunk 11]
```diff
@@ -0,0 +1,930 @@
...
+* Query Data using the Web API - `+https://msdn.microsoft.com/en-us/library/gg334767.aspx+`
+
+
+==== Parameters
+
+[%header%autowidth.spread]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | The name of the configuration to use. | | x
+| Datasense Query a| String |  The DSQL query that is going to use for retrieve. The query will be transformed in a retrieve URL internally. |  | x
```
Comment (valkyrie69) [inline microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc]:
```suggestion
| Datasense Query a| String |  The DSQL query to use to retrieve. The query is transformed in a retrieve URL internally. |  | x
```

File: microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc [hunk 12]
```diff
@@ -0,0 +1,930 @@
...
+
+==== Parameters
+
+[%header%autowidth.spread]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | The name of the configuration to use. | | x
+| Logical Name a| String |  The logical name of the entity. It is always the schema name in lower case. |  | x
+| Entities Attributes a| Array of Object |  A list with the attributes of the entities that are going to be updated as a List of Maps. |  `#[payload]` |
+| Use Single Transaction a| Boolean |  Use a single transaction for update. If the update fails, the transaction is rolled-back. |  false |
```
Comment (valkyrie69) [inline microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc]:
```suggestion
| Use Single Transaction a| Boolean |  Use a single transaction for an update. If the update fails, the transaction is rolled back. |  false |
```

File: microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc [hunk 13]
```diff
@@ -0,0 +1,930 @@
...
+
+==== Parameters
+
+[%header%autowidth.spread]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | The name of the configuration to use. | | x
+| Since a| String |  The required date format is `'yyyy-MM-dd'T'HH:mm:ss'Z'` |  |
+| Entity Type a| String | Describes business data. Example values are `account` and `contact`, which are defined in the Microsoft Dynamics 365 instance. |  | x
+| Primary Node Only a| Boolean |  Whether this source should be executed on the primary node only when running in a cluster. |  |
```
Comment (valkyrie69) [inline microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc]:
```suggestion
| Primary Node Only a| Boolean |  Whether to execute this source on only the primary node when running in a cluster. |  |
```

File: microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc [hunk 14]
```diff
@@ -0,0 +1,930 @@
...
+
+==== Parameters
+
+[%header%autowidth.spread]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | The name of the configuration to use. | | x
+| Since a| String |  The required date format is `'yyyy-MM-dd'T'HH:mm:ss'Z'` |  |
+| Entity Type a| String | Describes business data. Example values are `account` and `contact`, which are defined in the Microsoft Dynamics 365 instance.  |  | x
+| Primary Node Only a| Boolean |  Whether this source should only be executed on the primary node when running in a cluster. |  |
```
Comment (valkyrie69) [inline microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc]:
```suggestion
| Primary Node Only a| Boolean |  Whether to execute this source on only the primary node when running in a cluster. |  |
```

File: microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc [hunk 15]
```diff
@@ -0,0 +1,930 @@
...
+
+[[RedeliveryPolicy]]
+=== Redelivery Policy
+
+[%header%autowidth.spread]
+|===
+| Field | Type | Description | Default Value | Required
+| Max Redelivery Count a| Number | The maximum number of times a message can be redelivered and processed unsuccessfully before triggering a process-failed message |  |
+| Use Secure Hash a| Boolean | Whether to use a secure hash algorithm to identify a redelivered message. |  |
+| Message Digest Algorithm a| String | The secure hashing algorithm to use. If not set, the default is `SHA-256`. |  |
```
Comment (valkyrie69) [inline microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc]:
```suggestion
| Message Digest Algorithm a| String | The secure hashing algorithm to use.  | `SHA-256` |
```

File: microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc [hunk 16]
```diff
@@ -0,0 +1,930 @@
...
+| Successful a| Boolean |  |  |
+|===
+
+[[repeatable-in-memory-iterable]]
+=== Repeatable In Memory Iterable
+
+[%header%autowidth.spread]
+|===
+| Field | Type | Description | Default Value | Required
+| Initial Buffer Size a| Number | The number of instances to keep in memory to consume the stream and provide random access to it. If the stream contains more data than can fit into this buffer, then it expands according to the bufferSizeIncrement attribute, with an upper limit of Max In Memory Size. Default value is 100 instances. |  |
```
Comment (valkyrie69) [inline microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc]:
```suggestion
| Initial Buffer Size a| Number | The number of instances to keep in memory to consume the stream and provide random access to it. If the stream contains more data than can fit into this buffer, then it expands according to the bufferSizeIncrement attribute, with an upper limit of Max In Memory Size. Default value is 100 instances. | 100  |
```

File: microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc [hunk 17]
```diff
@@ -0,0 +1,930 @@
...
+| Buffer Size Increment a| Number | By how much the buffer size by expands if it exceeds its initial size. Setting a value of zero or lower means that the buffer should not expand, meaning that a STREAM_MAXIMUM_SIZE_EXCEEDED error is raised when the buffer gets full. Default value is 100 instances. |  |
```
Comment (valkyrie69) [inline microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc]:
```suggestion
| Buffer Size Increment a| Number | Specifies how much the buffer size expands if it exceeds its initial specified size. Setting a value of zero or lower means that the buffer should not expand, and that a STREAM_MAXIMUM_SIZE_EXCEEDED error is raised when the buffer gets full. Default value is 100 instances. |100  |
```

File: microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc [hunk 18]
```diff
@@ -0,0 +1,930 @@
...
+| Max Buffer Size a| Number | The maximum amount of memory to use. If more memory is used, a STREAM_MAXIMUM_SIZE_EXCEEDED error is raised. A value lower or equal to zero means no limit. |  |
```
Comment (valkyrie69) [inline microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc]:
```suggestion
| Max Buffer Size a| Number | The maximum amount of memory to use. If more memory is used, a STREAM_MAXIMUM_SIZE_EXCEEDED error is raised. A value of zero or lower means no limit. |  |
```

File: microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc [hunk 19]
```diff
@@ -0,0 +1,930 @@
...
+|===
+
+[[repeatable-file-store-iterable]]
+=== Repeatable File Store Iterable
+
+[%header%autowidth.spread]
+|===
+| Field | Type | Description | Default Value | Required
+| Max In Memory Size a| Number | The maximum amount of instances that are kept in memory. If more instances are required, then the connector starts to buffer the content on disk. |  |
```
Comment (valkyrie69) [inline microsoft-dynamics-365/2.4/modules/ROOT/pages/microsoft-dynamics-365-connector-reference.adoc]:
```suggestion
| Max In Memory Size a| Number | The maximum number of instances that are kept in memory. If more instances are required, then the connector starts to buffer the contents onto disk. |  |
```

**Review by valkyrie69 — DISMISSED** [review]

**Review by valkyrie69 — APPROVED** [review]

**Pull request merged**